paddle/fluid/operators/batch_norm_mkldnn_op.cc

```diff
@@ -110,6 +217,14 @@ class BatchNormMKLDNNOpKernel : public paddle::framework::OpKernel<T> {
     PADDLE_ENFORCE(scale_tz.size() == 1, "Dims of scale tensor is NOT 1");
     const unsigned int ic = scale_tz[0];
 
+    // MKLDNN requires a single piece of memory for scale and shift/bias data
+    const size_t scaleshift_size = 2 * ic;
```

Comment:
Can you tell me why we used 2?

Reply:
We have to preallocate one continuous piece of memory for 2 vectors, next we copy these vectors to scaleshift_data vector.